app/controllers/userController.js

```diff
@@ -27,19 +28,20 @@ let userControl = {
   // login user control
   loginUser: function (req, res) {
     if (!req.body.username) {
-      res.send({ Error: 'Username is required.' });
+      res.send({ error: 'Username is required.' });
       return;
     }
     if (!req.body.password) {
-      res.send({ Error: 'Password is required.' });
+      res.send({ error: 'Password is required.' });
       return;
     }
     models.User.findOne({ where: { username: req.body.username } })
       .then((user) => {
-        if (user.password = req.body.password) {
-          let token = createToken({ id: user.id, username: user.username, RoleId: user.RoleId });
+        if (bcrypt.compareSync(req.body.password, user.password)) {
+          console.log('Hola');
+          const token = createToken({ id: user.id, username: user.username, RoleId: user.RoleId });
```

Comment:
**Nitpick - Code quality**
Signing the entire user object might not be the best decision. Correct me if I'm wrong. What you need to sign is the Id and then define a function that extracts the user details you want to make available to the client